hooks/pre_gen_project.py

```diff
@@ -0,0 +1,32 @@
+'''
```

Comment:
Should this be like following instead of making the whole file a string?
```py
# Setting easy to read/maintain cookiecutter variables:

# -----------
# Web/API Variant
# cookiecutter._web_variant
# cookiecutter._api_variant
# -----------
if '{{ cookiecutter.variant }}' in ['Web only', 'Both']:
    '{{ cookiecutter.update({ '_web_variant': 'y' }) }}'

# API Variant
if '{{ cookiecutter.variant }}' in ['API only', 'Both']:
    '{{ cookiecutter.update({ '_api_variant': 'y' }) }}'

# -----------
# CSS Addons
# cookiecutter._bootstrap_addon
# cookiecutter._tailwinds_addon
# -----------

# Only project with web:
if '{{ cookiecutter._web_variant }}' == 'y':
    if '{{ cookiecutter.css_addon }}' == 'Bootstrap':
        '{{ cookiecutter.update({ '_bootstrap_addon': 'y' }) }}'
    elif '{{ cookiecutter.css_addon }}' == 'Tailwinds':
        '{{ cookiecutter.update({ '_tailwinds_addon': 'y' }) }}'
```

Reply:
Interesting question! :)
I've just tried and it does not work. Indeed, the templating commands are calling at a different time than the python script. 
So what happened in order is this:
1. All template instructions are executed, their value is replaced in the hook script:
    - `if` conditions gets the string values for choice variables ✅
    - BUT at that time, the update statement get executed too, setting all private variables to 'y' ⛔️
2. The hook script get executed: It run a bunch of if/else condition that will trigger statement with basic strings: No update in cookiiecutter variables can be done at that time. 

So at least we have to keep using the if/elif conditions from within the template execution ( `{% if [...] %} ` ). Eventually we can change `'''` by single line comments... The point here is just to ensure it won't impact the rest of the hook script ;-)

Reply:
Thanks for the clarification. If this is an issue then we can go with your solution 😉

Reply:
I think it's not an issue and Ok as is ;-) Let's see if Hoang think it is~